source/platforms/gitlab/GitLabAPI.ts

```diff
@@ -97,18 +102,18 @@ class GitLabAPI {
 
   getMergeRequestCommits = async (): Promise<GitLabMRCommit[]> => {
     this.d("getMergeRequestCommits", this.repoMetadata.repoSlug, this.repoMetadata.pullRequestID)
-    const commits: GitLabMRCommit[] = await this.api.MergeRequests.commits(
```

Comment:
These changes aren't great - having the type reference on the left means the type system checks to validate, having them on the right means there's no type safety anymore. Are you sure this is what you want?

Reply:
Originally the gitlab npm module's types were not being used:
(currently in master)
https://github.com/danger/danger-js/blob/04c33453b3216b24718df571235064bbc7e659ab/source/platforms/gitlab/GitLabAPI.ts#L55
https://github.com/danger/danger-js/blob/04c33453b3216b24718df571235064bbc7e659ab/source/platforms/gitlab/GitLabAPI.ts#L61
So, all the method results had return type `any` which would get casted to whatever we put on the left side.

I changed it to use the npm module types:
https://github.com/rohit-gohri/danger-js/blob/13fe5ebe38407e2ff4abe491e5340c0fe4d07ee8/source/platforms/gitlab/GitLabAPI.ts#L57
```private api: InstanceType<typeof Gitlab>```

This way we at least we get type checks for the module's methods. But the module's type definitions define all the response types as `Promise<object | object[]`> and that doesn't get cast automatically hence the need for `as GitLabMRCommit[]`. So overall we are gaining more type safety (any deprecated methods would break build instead of at runtime).

I hope this makes sense.

Reply:
It does - thanks